### PR TITLE
Instructions for adding a container disk when creating a VM

### PR DESCRIPTION
**Pull request by LucasSaintarbor**
**Summary**
Resolves https://github.com/harvester/harvester/issues/3791

- Add instructions for adding a container disk 
- Add context for adding a container disk
- Reference https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#custom-disk-image-path

**Next steps**
- Adding to v1.1

**Questions**
- How to get a working example of - [registry.suse.com/suse/vmdp/vmdp](http://registry.suse.com/suse/vmdp/vmdp)

### CODE REVIEW

File: docs/vm/create-vm.md [hunk 1]
```diff
@@ -52,13 +52,28 @@ A volume's [StorageClass](../advanced/storageclass.md) can be specified when add
...
+1. Go to the **Volumes** tab.
+2. Select **Add Container**.
+
+![add-container-volume](/img/v1.2/vm/add-container-volume-1.png)
+
+3. Enter a **Name** for the container disk.
+4. Choose a disk **Type**.
+5. Add a **Docker Image**.
+    - Images must be placed into the `/disk` directory.
+    - Raw and qcow2 formats are supported, but qcow2 is recommended in order to reduce the container image's size. If you use an unsupport image format, the VM will get stuck in a `Running` state.
```
Comment (vickyhella) [inline docs/vm/create-vm.md:72]:
```suggestion
    - Raw and qcow2 formats are supported, but qcow2 is recommended in order to reduce the container image's size. If you use an unsupported image format, the VM will get stuck in a `Running` state. 
```

File: docs/vm/create-vm.md [hunk 2]
```diff
@@ -52,13 +52,28 @@ A volume's [StorageClass](../advanced/storageclass.md) can be specified when add
...
+2. Select **Add Container**.
+
+![add-container-volume](/img/v1.2/vm/add-container-volume-1.png)
+
+3. Enter a **Name** for the container disk.
+4. Choose a disk **Type**.
+5. Add a **Docker Image**.
+    - Images must be placed into the `/disk` directory.
+    - Raw and qcow2 formats are supported, but qcow2 is recommended in order to reduce the container image's size. If you use an unsupport image format, the VM will get stuck in a `Running` state. 
+    - A contaier disk also allows you to store disk images in any folder in the `/disk` directory by adding a [custom disk image path](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#custom-disk-image-path).
```
Comment (vickyhella) [inline docs/vm/create-vm.md:73]:
```suggestion
    - A container disk also allows you to store disk images in any folder in the `/disk` directory by adding a [custom disk image path](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#custom-disk-image-path).
```

File: docs/vm/create-vm.md [hunk 3]
```diff
@@ -52,13 +52,28 @@ A volume's [StorageClass](../advanced/storageclass.md) can be specified when add
 
 ![create-vm](/img/v1.2/vm/create-vm-volumes.png)
 
-:::info Container Disk
+### Adding a container disk
 
-`Container disks` are ephemeral storage devices that can be assigned to any number of VMs. This makes them an ideal tool for users who want to replicate a large number of VM workloads or inject machine drivers that do not require persistent data.
+A container disk is an ephemeral storage volume that can be assigned to any number of VMs and provides the ability to store and distribute VM disks in the container image registry.
```
Comment (vickyhella) [inline docs/vm/create-vm.md:57]:
```suggestion
A container disk is an ephemeral storage volume that can be assigned to any number of VMs and provides the ability to store and distribute VM disks in the container image registry. A container disk is:
```

File: docs/vm/create-vm.md [hunk 4]
```diff
@@ -52,13 +52,28 @@ A volume's [StorageClass](../advanced/storageclass.md) can be specified when add
...
+- An ideal tool for users who want to replicate a large number of VM workloads or inject machine drivers that do not require persistent data. Ephemeral volumes are designed for VMs that need more storage but don't care whether that data is stored persistently across VM restarts or only expect some read-only input data to be present in files, like configuration data or secret keys.
```
Comment (vickyhella) [inline docs/vm/create-vm.md:58]:
```suggestion
- An ideal tool if you want to replicate a large number of VM workloads or inject machine drivers that do not require persistent data. Ephemeral volumes are designed for VMs that need more storage but don't care whether that data is stored persistently across VM restarts or only expect some read-only input data to be present in files, like configuration data or secret keys.
```

**Review by vickyhella — COMMENTED** [review]
Just some small suggestions, the rest LGTM!

**Comment by LucasSaintarbor** [comment]
> Just some small suggestions, the rest LGTM!

Thanks @vickyhella for reviewing! Added your suggestions

**Review by vickyhella — APPROVED** [review]

**Comment by vickyhella** [comment]
@LucasSaintarbor Please also update the v1.1 doc as mentioned in this PR's description, and then this PR is good to merge

**Comment by LucasSaintarbor** [comment]
@bk201 May you review this update when you get the chance? Resolves: https://github.com/harvester/harvester/issues/3791

File: docs/vm/create-vm.md [hunk 5]
```diff
@@ -52,13 +52,25 @@ A volume's [StorageClass](../advanced/storageclass.md) can be specified when add
...
+
+A container disk is added when creating a VM by providing a Docker image. When creating a VM, follow these steps:
+
+1. Go to the **Volumes** tab.
+1. Select **Add Container**.
+  ![add-container-volume](/img/v1.2/vm/add-container-volume-1.png)
+1. Enter a **Name** for the container disk.
+1. Choose a disk **Type**.
+1. Add a **Docker Image**.
+    - Images must be placed into the `/disk` directory.
```
Comment (bk201) [inline docs/vm/create-vm.md:69]:
```suggestion
    - A disk image (in qcow2 or raw format) must be placed into the `/disk` directory.
```

Thanks, @LucasSaintarbor; we need to clarify a bit more here. The user uses a **disk image** (qcow2 or raw format, like `/disk/ubuntu.qcow2`) inside a **container image** as VM's disk. Please help adjust the wording if it doesn't sound right.

File: docs/vm/create-vm.md [hunk 6]
```diff
@@ -52,13 +52,25 @@ A volume's [StorageClass](../advanced/storageclass.md) can be specified when add
...
+    - Raw and qcow2 formats are supported, but qcow2 is recommended in order to reduce the container image's size. If you use an unsupported image format, the VM will get stuck in a `Running` state.
+    - A container disk also allows you to store disk images in any folder in the `/disk` directory by adding a [custom disk image path](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#custom-disk-image-path).
```
Comment (bk201) [inline docs/vm/create-vm.md:71]:
```suggestion
    - A container disk also allows you to store disk images in the `/disk` directory. An example of creating such a container image can be found [here](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#containerdisk-workflow-example). 
```

We can suggest the formal way here (to put disk image in `/disk`). `customer-disk-image-path` is for user to specify a different directory to contain the disk image and it won't work with our GUI.

**Comment by LucasSaintarbor** [comment]
@bk201 just want to confirm if this is okay to merge? Thanks!

**Comment by bk201** [comment]
> @bk201 just want to confirm if this is okay to merge? Thanks!

Sure, LGTM. Apologize for the delay.

**Review by bk201 — APPROVED** [review]

**Pull request merged**